src/app/shared/services/http.service.ts

```diff
@@ -21,22 +21,13 @@ export class HttpService {
   }
 
   post(url, values): Observable<any> {
-    return this.http.post(this.serverAddress + url, values, {observe: 'response'});
+    return this.http.post(this.serverAddress + url, values, {observe: 'response'}).map(data => data.body);
```

Comment:
This might lead to serious problems, as we have used `data = data.body;` all over the project till now :))